rust/processor/src/models/token_models/ans_lookup.rs

```diff
@@ -74,107 +77,225 @@ impl OptionalString {
-                                        .new_address
-                                        .get_string()
-                                        .map(|s| standardize_address(&s)),
-                                    last_transaction_version: txn_version,
-                                    expiration_timestamp,
-                                    token_name,
-                                }
-                            },
-                            ANSEvent::RegisterNameEventV1(inner) => {
-                                let expiration_timestamp = parse_timestamp_secs(
-                                    bigdecimal_to_u64(&inner.expiration_time_secs),
-                                    txn_version,
-                                );
-                                let subdomain =
-                                    inner.subdomain_name.get_string().unwrap_or_default();
-                                let mut token_name = format!("{}.apt", &inner.domain_name);
-                                if !subdomain.is_empty() {
-                                    token_name = format!("{}.{}", &subdomain, token_name);
-                                }
-                                Self {
-                                    domain: inner.domain_name,
-                                    subdomain,
-                                    registered_address: None,
-                                    last_transaction_version: txn_version,
-                                    expiration_timestamp,
-                                    token_name,
-                                }
-                            },
-                        };
-
-                        current_ans_lookups.insert(
-                            (
-                                current_ans_lookup.domain.clone(),
-                                current_ans_lookup.subdomain.clone(),
-                            ),
-                            current_ans_lookup,
-                        );
+                    },
+                    _ => Ok(None),
+                } {
+                    if let Some(name_record) = maybe_name_record {
+                        current_ans_lookups
+                            .entry((name_record.domain.clone(), name_record.subdomain.clone()))
+                            .and_modify(|e| {
+                                e.registered_address = name_record.registered_address.clone();
+                                e.expiration_timestamp = name_record.expiration_timestamp;
+                                e.is_primary = name_record.is_primary || e.is_primary;
+                                e.is_deleted = name_record.is_deleted;
+                            })
+                            .or_insert(name_record);
                     }
+                } else {
+                    continue;
                 }
             }
         }
+
         current_ans_lookups
     }
+
+    pub fn from_write_table_item(
```

Comment:
Add comment.

Reply:
If possible can you elaborate a bit more about how the parsing actually works. For example, something like 
We have 2 tables, one table the key is nameRecord, the value is token, the other table the key is address and value is nameRecord. 

In case 1: this happens
case2: this happens.